### DISCUSSION TITLE
Getting points from mesh

### DISCUSSION THREAD

**Discussion by marcodc-sys** [pyvista/pyvista#2524] [Q&A]
Dear all,
my purpose is to interactively select cells from a mesh (STL file) inside a polygon drawn with the mouse. I tried enable_cell_picking() but the polygon is a rectangle. I would like to define the polygon drawing it with the mouse and pick all the cells inside that polygon.
Thanks a lot.

**Comment by marcodc-sys** [comment]
Sorry, can someone answer me?

**Comment by banesullivan** [comment]
PyVista currently only supports a rectangular selection in this style. 

You could however implement a custom variant of `enable_cell_picking` that uses a few of the different event handlers to capture different click events on the screen to draw the polygon in the display space, then perform the hardware picking after the polygon is closed.

This is a rather involved process and I so wish I had the time to develop a full example for you... sorry we couldn't be of more help.

**Reply by marcodc-sys** [reply]
Thanks a lot, it would be great having an example. Thanks
